File: services/user-service/src/main/java/com/mealmap/userservice/core/dto/user/UserDto.java
package com.mealmap.userservice.core.dto.user;

import com.mealmap.userservice.entity.enums.Role;
import com.mealmap.userservice.entity.value.UserStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private UUID id;

    private String phoneNumber;

    private String email;

    private String firstName;

    private String lastName;

    private Integer organizationId;

    private UserStatus status;

    private List<Role> roles;

    private LocalDate createdAt;
}

File: frontend/src/api/user/dto/UserDto.ts
import { Role } from "../enums/Role";
import { UserStatus } from "./UserStatus";

export interface UserDto {
  id: string; // UUID as a string
  phoneNumber: string;
  email: string;
  firstName: string;
  lastName: string;
  organizationId: number;
  status: UserStatus;
  roles: Role[];
  createdAt: string; // ISO date string
}
